### DISCUSSION TITLE
DCS Protocol and Auth

### DISCUSSION THREAD

**Discussion by cappelaere** [opendcs/opendcs#148] [Q&A]
I am looking at the DCP Data Service Protocol V14 2/29/2016
I am trying to connect to the LRGS Server in authenticated mode.
p13 it says: 
AuthHelloBody ::= username SP time SP AuthenticatorHash

However, LddsClient:java :346 seems to add + " " + DdsVersion.getVersion();
Do I have an old version of protocol spec?

I am using: decj lrgs.lddc.Client to test the interface.  Seems to work when I send the auth command and then status command.
Stupid question: Is there a way to turn DEBUG on?  I tried -d 3... but no cigar.
Thanks,
Pat.

**Comment by cappelaere** [comment]
I was able to connect in authenticated mode to the server from client (written in javascript).  I had to use SHA1.  I have to look into this more... but I can keep going.  Thanks.
And I can get the status...
So, I could have this client be my API to prometheus and others.... Cool!

**Reply by MikeNeilson** [reply]
Whelp, SHA1, that'll need to go soon. But anyways, please share what you've created as soon as possible. I was going over the LRGS archive code this weekend and while stable will require some major changes. Having your docker container and the tests your doing (especially if by some chance you could get those at least structured into Junit5 test, but that's just a dream.) would make that work go much faster since I wouldn't have to duplicate it.

I know what you're doing is a proof of concept, but it will still be a good starting point for improvements.

**Reply by MikeNeilson** [reply]
For completion of the discussion. SHA256 is supported, but looking at the client code it only gets used after the server response with a failure message.

**Comment by cappelaere** [comment]
Thanks for looking into this.
I was under the impression that SHA-256 was required for DDS v14.  It does not seem so.
Is the curent spec up to date?  what version should I use?

Beyond user login, what are the data security requirements for DCS?
Any authenticated user can have access to all the data?
Is there any expectation of sensitive/proprietary information ?
Thanks,
Pat.

**Reply by MikeNeilson** [reply]
Thinking back it's supposed to be, so I'm sure why you were having issues, but this is the first I've thoroughly dug into the LRGS code itself. As for the spec, I just started documenting the protocol as I realized after our discussed, it hadn't actually been done yet.

As for data security, at this time there is no expectation of sensitive or proprietary data; if you think that may be a need though add it to here: https://github.com/opendcs/opendcs/discussions/152
